tests/testthat/test_PredictionSurv.R

```diff
@@ -196,19 +196,19 @@ test_that("filtering", {
   expect_set_equal(p2$data$row_ids, c(20, 37, 42))
   expect_set_equal(p3$data$row_ids, c(20, 37, 42))
   expect_set_equal(p4$data$row_ids, c(20, 37, 42))
-  expect_numeric(p$data$crank, any.missing = FALSE, len = 3)
-  expect_numeric(p2$data$crank, any.missing = FALSE, len = 3)
-  expect_numeric(p3$data$crank, any.missing = FALSE, len = 3)
-  expect_numeric(p4$data$crank, any.missing = FALSE, len = 3)
-  expect_numeric(p$data$lp, any.missing = FALSE, len = 3)
-  expect_numeric(p2$data$lp, any.missing = FALSE, len = 3)
-  expect_numeric(p3$data$lp, any.missing = FALSE, len = 3)
-  expect_numeric(p4$data$lp, any.missing = FALSE, len = 3)
-  expect_matrix(p$data$distr, nrows = 3)
+  expect_numeric(p$data$crank, any.missing = FALSE, len = 3L)
+  expect_numeric(p2$data$crank, any.missing = FALSE, len = 3L)
+  expect_numeric(p3$data$crank, any.missing = FALSE, len = 3L)
+  expect_numeric(p4$data$crank, any.missing = FALSE, len = 3L)
+  expect_numeric(p$data$lp, any.missing = FALSE, len = 3L)
+  expect_numeric(p2$data$lp, any.missing = FALSE, len = 3L)
+  expect_numeric(p3$data$lp, any.missing = FALSE, len = 3L)
+  expect_numeric(p4$data$lp, any.missing = FALSE, len = 3L)
+  expect_matrix(p$data$distr, nrows = 3L)
   expect_array(p2$data$distr, d = 3)
-  expect_equal(nrow(p2$data$distr), 3)
-  expect_true(inherits(p3$data$distr, "Matdist"))
-  expect_true(inherits(p4$data$distr, "Arrdist"))
+  expect_identical(nrow(p2$data$distr), 3L)
+  expect_s3_class(p3$data$distr, "Matdist")
```

Comment:
strange that this passes the test, `Matdist` objects have classes: `"Matdist"       "SDistribution" "Distribution"  "R6"`, so better tp have `expect_r6(...)`?

Reply:
you're correct its adjusted with some more explicit class checks in other places